modules/dnn/src/webgpu/src/buffer.cpp

```diff
@@ -33,23 +33,27 @@ void Buffer::setBufferData(const void * data, size_t size)
 
 const void* Buffer::MapReadAsyncAndWait() 
 {
-    wgpu::BufferDescriptor desc = {};
-    desc.size = size_;
-    desc.usage = wgpu::BufferUsage::CopyDst | wgpu::BufferUsage::MapRead;
-    wgpu::Buffer gpuReadBuffer = device_->CreateBuffer(& desc);
-    wgpu::CommandEncoder encoder = device_->CreateCommandEncoder();
-    encoder.CopyBufferToBuffer( buffer_, 0, 
-                                gpuReadBuffer, 0, size_);
-    wgpu::CommandBuffer cmdBuffer = encoder.Finish();
-    wQueue->Submit(1, &cmdBuffer);
-    gpuReadBuffer.MapReadAsync(BufferMapReadCallback, this);
+    if(! gpuReadBuffer_)
+    {
+        wgpu::BufferDescriptor desc = {};
+        desc.size = size_;
+        desc.usage = wgpu::BufferUsage::CopyDst | wgpu::BufferUsage::MapRead;
+        gpuReadBuffer_ = device_->CreateBuffer(& desc);
+        wgpu::CommandEncoder encoder = device_->CreateCommandEncoder();
+        encoder.CopyBufferToBuffer( buffer_, 0, 
+                                    gpuReadBuffer_, 0, size_);
+        wgpu::CommandBuffer cmdBuffer = encoder.Finish();
+        encoder.Release();
+        wQueue->Submit(1, &cmdBuffer);
+        cmdBuffer.Release();
+    }
```

Comment:
the staging buffer create and copy should be seperated.

create can only do once, yet copy should be conducted every time this function be called.

Reply:
> the staging buffer create and copy should be seperated.
> 
> create can only do once, yet copy should be conducted every time this function be called.

Thanks, I've modified it.